packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -46,6 +50,38 @@ export function trackNetworkError(configuration: LogsConfiguration, errorObserva
   }
 }
 
+// TODO: ideally, computeResponseData should always call the `callback` with a string instead of
+// `any`. But to keep retrocompatibility, in the case of XHR with a `responseType` different than
+// "text", the response data should be whatever `xhr.response` is. This is a bit confusing as Logs
+// event 'stack' is expected to be a string. This should be changed in a future major version as it
+// could be a breaking change.
+export function computeResponseData(
+  { xhr, response, error }: { xhr?: XMLHttpRequest; response?: Response; error?: unknown },
+  configuration: LogsConfiguration,
+  callback: (responseData?: any) => void
+) {
+  if (xhr) {
+    if (typeof xhr.response === 'string') {
+      callback(truncateResponseText(xhr.response, configuration))
+    } else {
+      callback(xhr.response)
+    }
+  } else if (response) {
+    response
+      .clone()
+      .text()
+      .then(
+        (text) => callback(text),
+        (error) => callback(`Unable to retrieve response: ${error as string}`)
+      )
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏I am not sure if it was working fine with previous async/await version but should not we wrap the callback with a `monitor()`?

Reply:
Indeed! I'll wrap it in a `monitor()`.